### PR TITLE
Add the ability to list VINs with a specific package installed.

### PR DESCRIPTION
**Pull request by stevana**
Requested by @alexhumphreys.

### CODE REVIEW

**Comment by alexhumphreys** [comment]
Looks OK to be, cheers!

File: external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDAO.scala [hunk 1]
```diff
@@ -29,4 +29,10 @@ object VehicleDAO {
   def list: DBIO[Seq[Vehicle]] =
     vehicles.result
 
+  def exists(vin: Vehicle.Vin): DBIO[Option[Vehicle]] =
```
Comment (koshelev) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDAO.scala]: I would prefer `load`. Exists implies `boolean` result

Comment (stevana) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDAO.scala]: Only if you are a classical logician, I'm a constructivist -- exists provides the witness.

File: external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDirectives.scala [hunk 2]
```diff
@@ -37,7 +40,16 @@ class VehicleDirectives(implicit db: Database, mat: ActorMaterializer, ec: Execu
     pathPrefix("vehicles") {
       get {
         pathEnd {
-          complete(db.run(VehicleDAO.list))
+          parameter('package.as[Package.NameVersion].?) {
+            case Some(nameVersion) =>
+              val packageName   : Package.Name    = Refined(nameVersion.get.split("-").head)
```
Comment (koshelev) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDirectives.scala]: I don't think it is a good idea to parse name and version. Name also can contain `-`. What is a reason to pass it as a single string anyway?

File: external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleFunctions.scala [hunk 3]
```diff
@@ -73,10 +71,32 @@ object VehicleFunctions {
     for {
       _  <- exists(vin)
       ps <- packagesOnVinMap
-      .map(_
-        .get(vin)
-        .toList
-        .flatten)
+              .map(_
```
Comment (koshelev) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleFunctions.scala]: Single line, please

File: external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleFunctions.scala [hunk 4]
```diff
@@ -73,10 +71,32 @@ object VehicleFunctions {
...
+                .get(vin)
+                .toList
+                .flatten)
     } yield ps
 
+  def vinsThatHavePackageMap
+    (implicit db: Database, ec: ExecutionContext)
+      : Future[Map[Package.Id, Seq[Vehicle.Vin]]] =
+    db.run(InstalledPackages.list)
```
Comment (koshelev) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleFunctions.scala]: Single line, please

File: external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleFunctions.scala [hunk 5]
```diff
@@ -73,10 +71,32 @@ object VehicleFunctions {
...
+        .mapValues(_.map(_._1)))
+
+  def vinsThatHavePackage
+    (pkgId: Package.Id)
+    (implicit db: Database, ec: ExecutionContext)
+      : Future[Seq[Vehicle.Vin]] =
+    for {
+      _  <- existsPackage(pkgId)
+      vs <- vinsThatHavePackageMap
+              .map(_
```
Comment (koshelev) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleFunctions.scala]: This also passes to a single line

File: external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDirectives.scala (hunk 2 above)
Comment (stevana) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDirectives.scala]: Yeah, I agree this is bad. The same thing is done for package filters... It's on my to do list...

File: external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDAO.scala (hunk 1 above)
Comment (koshelev) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDAO.scala]: I have no idea what all those words mean, so let's use `load` :)

File: external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDirectives.scala (hunk 2 above)
Comment (koshelev) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDirectives.scala]: Would you please create an Jira task and reference it here?

File: external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDAO.scala (hunk 1 above)
Comment (stevana) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDAO.scala]: In classical logic a proposition is either true or false, i.e. boolean. In constructive logic, e.g. type theory, you need to construct the object (the term) that witnesses the truth of the proposition (the type).

It's a fascinating topic: https://en.wikipedia.org/wiki/Constructivism_(mathematics)

File: external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDirectives.scala (hunk 2 above)
Comment (stevana) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDirectives.scala]: https://advancedtelematic.atlassian.net/browse/SOTA-139

File: external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDAO.scala (hunk 1 above)
Comment (koshelev) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDAO.scala]: Sarcasm

**Comment by koshelev** [comment]
please squash commits

**Comment by stevana** [comment]
Squashed.

**Pull request merged**